browser/main/lib/dataApi/createNoteFromUrl.js

```diff
@@ -0,0 +1,90 @@
+const http = require('http')
+const https = require('https')
+const { createTurndownService } = require('../../../lib/turndown')
+const createNote = require('./createNote')
+
+import { push } from 'connected-react-router'
+import ee from 'browser/main/lib/eventEmitter'
+
+function validateUrl (str) {
+  if (/^(?:(?:(?:https?|ftp):)?\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})).?)(?::\d{2,5})?(?:[/?#]\S*)?$/i.test(str)) {
+    return true
+  } else {
+    return false
+  }
+}
+
+function createNoteFromUrl (url, storage, folder, dispatch = null, location = null) {
+  return new Promise((resolve, reject) => {
+    const td = createTurndownService()
+
+    if (!validateUrl(url)) {
+      reject({result: false, error: 'Please check your URL is in correct format. (Example, https://www.google.com)'})
+    }
+
+    const request = url.startsWith('https') ? https : http
+
+    const req = request.request(url, (res) => {
+      let data = ''
+
+      res.on('data', (chunk) => {
+        data += chunk
+      })
+
+      res.on('end', () => {
+        const html = document.createElement('html')
+        html.innerHTML = data
+
+        const scripts = html.getElementsByTagName('script')
```

Comment:
@AWolf81 Instead of doing this, can we use the turndown remove rule?
https://github.com/domchristie/turndown#removefilter

Reply:
@zerox-DG good idea. I'll change this in one week as I'm without a computer at the moment.

`remove` filter should simplify this.

Reply:
@ZeroX-DG OK, I've changed the filtering. I've created a  basic Codesandbox example to check that the `script` tag is not available in Bootstnote.

https://4bckd.csb.app/

If you open the URL in the browser it will execute an `alert` and in BoostNote there is no `script` tag available.
Is there more I can test?

The `html` constant can be removed as Turndown is handling the conversion string to md - it was just there for the script filtering.

Reply:
I think it looks good now. Thank you for your contribution!